File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-10.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<int> set = new HashSet<int>();

        int l = 0;
        int max = 0;

        for(int r = 0; r < s.Length; r++){
            while(set.Contains(s[r])){
                set.Remove(s[l]);
                l++;
            }
            max = Math.Max(max, r - l + 1);
            set.Add(s[r]);
        }
        return max;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-10.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0;
        int profit = 0;
        for(int r = 0; r < prices.length; r++){
            profit = Math.max(profit, prices[r] - prices[l]);

            if(prices[l] > prices[r]){
                l = r;
            }
        }
        return profit;
    }
}
